presto-main/src/main/java/com/facebook/presto/operator/WorkProcessorUtils.java

```diff
@@ -45,27 +44,18 @@ private WorkProcessorUtils() {}
             @Override
             protected T computeNext()
             {
-                while (true) {
-                    if (processor.process()) {
-                        if (processor.isFinished()) {
-                            return endOfData();
-                        }
-
-                        return processor.getResult();
-                    }
-                    else if (processor.isBlocked()) {
-                        try {
-                            processor.getBlockedFuture().get();
-                        }
-                        catch (InterruptedException e) {
-                            Thread.currentThread().interrupt();
-                            throw new RuntimeException(e);
-                        }
-                        catch (ExecutionException e) {
-                            throw new RuntimeException(e);
-                        }
+                if (processor.process()) {
+                    if (processor.isFinished()) {
+                        return endOfData();
                     }
+
+                    return processor.getResult();
+                }
+                else if (processor.isBlocked()) {
+                    throw new IllegalStateException("Cannot iterate over blocking WorkProcessor");
                 }
+
+                throw new IllegalStateException("Cannot iterate over yielding WorkProcessor");
```

Comment:
move above as `checkState(processor.process(), "Cannot iterate over yielding WorkProcessor")` ?

Reply:
It can be blocked too when `process()` returns `false`